testsuite/mailhog.py

```diff
@@ -60,6 +66,19 @@ def _all_messages(self, start):
             count = response["count"]
         return response
 
+    def delete_searched_messages(self):
```

Comment:
My understanding is that this is a replacement of a "cleanup" already included in the test. If so i would rather suggest to abandon it. It wasn't best idea to implement in that way. Anyway mailhog requires different cleanup as there is lots of emails generated during execution of all tests. So in my opinion it is absolutely fine to remove that cleanup from the test without replacement.